app/main.py

```diff
@@ -159,7 +197,71 @@ def get_stops():
     }
 
     print(search_values)
+    query = build_query('stops', search_values.keys(), search_values)
+
+    return json.dumps([get_stop(*stop) for stop in sql.execQuery(query)]), {'Content-Type': 'application/json'}
+
+
+@app.route('/get-lines', methods=['GET'])
+def get_lines():
+    """Get lines from the MySQL DB."""
+    operator = request.args.get("operator", default="%", type=str).split(",")
+    internal_id = request.args.get("internal_id", default="%", type=str).split(",")
+    public_id = request.args.get("public_id", default="%", type=str).split(",")
+    # TODO: handle destination names with spaces.
+    name = request.args.get("name", default="%", type=str).split(",")
+    destination_name = request.args.get("destination", default="%", type=str).split(",")
+    direction = request.args.get("direction", default="1", type=int)
+    transport_type = request.args.get("transport_type", default="%", type=str).split(",")
+    total_stops = request.args.get("total_stops", default="%", type=int)
+
+    sql = MysqlConnector()
+
+    search_values = {
+            "operators.name": operator,
+            "internal_id": internal_id,
+            "public_id": public_id,
+            "transport_lines.name": name,
+            "destination_name": destination_name,
+            "direction": direction,
+            "transport_types.name": transport_type,
+            "total_stops": total_stops
+    }
+
+    join = {
+            "operators": ("transport_lines.operator_id", "operators.id"),
+            "transport_types": ("transport_lines.transport_type_id", "transport_types.id")
+    }
+
+    query = build_query("transport_lines", search_values.keys(), search_values, inner_join=join)
+    return json.dumps([get_transport_line(*line) for line in sql.execQuery(query)]), {"Content-Type": "application/json"}
+
+
+@app.route('/get-line-info', methods=['GET'])
+def get_line_info():
+    """Get the information for one or more specific lines, like the route
+       that line takes from the MySQL DB.
+    """
+    internal_id = request.args.get("internal_id", default="%", type=str).split(",")
+
+    sql = MysqlConnector()
+
+    line_id = sql.getId("transport_lines", {"internal_id": internal_id[0]})
+    stop_query = build_query("transport_lines_stops", ["id"], {"transport_line_id": [line_id]}) + "ORDER BY order_number"
+    stop_ids = list(sql.execQuery(stop_query))
+
+    #TODO Error handling! Not every stop_id has a corresponding stop for some reason.
+    stops = [sql.execQuery(build_query("stops", ["stop_code"], {"id": [stop_id[0]]}))[0][0] for stop_id in stop_ids]
```

Comment:
This is still very hacky and does require some rewriting/error handling, but it does the job for now and we need this for analytics.

Reply:
We'll fix it after the demo this week 👍